### PR TITLE
{ACR}: expose --expiration from token commands

### PR DESCRIPTION
**Pull request by yugangw-msft**
**Description**  
See history entries below

**History Notes**  
az acr token create: expose --expiration for accurate time  
az acr token credential generate: expose --expiration for accurate time  
az acr show-endpoints: warn if the dedicated endpoint is not enabled

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/acr/commands.py [hunk 1]
```diff
@@ -167,7 +167,7 @@ def load_command_table(self, _):  # pylint: disable=too-many-statements
         g.command('delete', 'acr_delete')
         g.show_command('show', 'acr_show')
         g.command('show-usage', 'acr_show_usage', table_transformer=usage_output_format)
-        g.command('show-endpoints', 'acr_show_endpoints', table_transformer=endpoints_output_format, is_preview=True)
```
Comment (yugangw-msft) [inline src/azure-cli/azure/cli/command_modules/acr/commands.py:170]: Do apologize for the back and forth, but we decide to make it GA

**Comment by yonzhan** [comment]
ACR

File: src/azure-cli/azure/cli/command_modules/acr/_params.py [hunk 2]
```diff
@@ -332,6 +333,11 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
         c.argument('expiration_in_days', options_list=['--expiration-in-days', c.deprecate(target='--days', redirect='--expiration-in-days', hide=True)],
                    help='Number of days for which the credentials will be valid. If not specified, the expiration will default to the max value "9999-12-31T23:59:59.999999+00:00"', type=int, required=False)
 
+    for scope in ['acr token create', 'acr token credential generate']:
+        with self.argument_context(scope) as c:
+            c.argument('expiration', validator=validate_expiration_time,
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/acr/_params.py:338]: seems there're different expiration unit/measure: expire in days, expire at exact date etc.

Comment (fengzhou-msft) [reply src/azure-cli/azure/cli/command_modules/acr/_params.py:338]: consider `expire_at` as option name?

Comment (yugangw-msft) [reply src/azure-cli/azure/cli/command_modules/acr/_params.py:338]: @yungezz, Yes `expiration` is for exact date, the `expiration-in-days` is for days to make the input easier.

@fengzhou-msft, I name it as `expiration` for wording consistency with `--expiration-in-days`; or I can do `--expire-in-days` and `--expire-at`, let me know.

**Comment by yugangw-msft** [comment]
@fengzhou-msft , the CI failures was not caused by this PR. Could you please cross check and merge when ready?

File: src/azure-cli/azure/cli/command_modules/acr/_params.py (hunk 2 above)
Comment (fengzhou-msft) [reply src/azure-cli/azure/cli/command_modules/acr/_params.py:338]: I think `--expire-in-days` and `--expire-at` is more clear.

**Comment by fengzhou-msft** [comment]
> @fengzhou-msft , the CI failures was not caused by this PR. Could you please cross check and merge when ready?

Can you merge the latest dev branch? It has the fix for the CI error.

**Comment by yugangw-msft** [comment]
> Can you merge the latest dev branch? It has the fix for the CI error.

@fengzhou-msft, done

File: src/azure-cli/azure/cli/command_modules/acr/_params.py (hunk 2 above)
Comment (fengzhou-msft) [reply src/azure-cli/azure/cli/command_modules/acr/_params.py:338]: @yugangw-msft let me know whether you would make the change. Either is fine as users can always know the meaning of the options from help messages.

**Comment by fengzhou-msft** [comment]
BTW, you don't need the colon if you don't have `BREAKING CHANGE` or put a command name in the title. See the title format [here](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

**Pull request merged**